nunaliit2-js/src/main/nunaliit-es6/src/n2es6/n2mapModule/N2MapCanvas.js

```diff
@@ -663,6 +664,51 @@ class N2MapCanvas {
 		//Everytime a change is detected. The N2CouchDbSource/N2ModelSource will be update
 		customMap.on('movestart', onMoveStart);
 
+		if(this.options.wmsGetFeatureInfo) {
+			customMap.on('singleclick', function(evt) {
+				const $dialogDivFind = $('body').find('#n2FeatureInfoDialog');
+				let $dialogDiv;
+				const calltime = Date.now();
+				const viewResolution = _this.n2View.getResolution();
+
+				if($dialogDivFind.length <= 0) {
+					$dialogDiv = $('<div>').attr('id', 'n2FeatureInfoDialog').attr('title', _loc('Feature Info')).attr('calltime', calltime).appendTo($('body'));
+				} else if($dialogDivFind[0].attributes.calltime.value != calltime) {
+					$dialogDiv = $($dialogDivFind[0]);
+					$dialogDiv.attr('calltime', calltime);
+					$dialogDiv.empty();
+					$dialogDiv.dialog('close');
+				} else {
+					$dialogDiv = $($dialogDivFind[0]);
+				}
+
+				
+				customMap.forEachLayerAtPixel(evt.pixel, function(layer) {
+					if(layer.values_.source instanceof TileWMS) {
+						const url = layer.values_.source.getFeatureInfoUrl(
+							evt.coordinate,
+							viewResolution,
+							'EPSG:3857',
+							{'INFO_FORMAT': 'application/json'}
+						);
+						if (url) {
+							fetch(url)
+							.then((response) => response.json())
+							.then((data) => {
+								if(data.features && data.features.length > 0) {
+									$('<p>').text(`${layer.values_.title}: ${data.features[0].properties.GRAY_INDEX}`).appendTo($dialogDiv);
```

Comment:
This isn't apparent whether or not it's data that comes with features normally. If it does not come with data normally in Nunaliit, it should be changed.

Reply:
I've put in a method that can be overriden and then moved the custom code to the project. Can you take a look at latest commit and see what you think?